### ISSUE TITLE
Error executing 'reshape' function

### ISSUE THREAD

**Issue by mdekstrand** [bedatadriven/renjin#128]
I am attempting to use 'reshape' to reshape a basic data frame with 2 ID variables. If I change it to use 1 ID variable, it also fails but in a different place in the 'reshape' implementation.

Transcript (source + output):

```
Using local repository: /Users/michael/.m2/repository
Renjin 0.7.1578
Copyright (C) 2013 The R Foundation for Statistical Computing
Copyright (C) 2014 BeDataDriven
> predictions = read.csv("predictions.csv")
> preds.wide = reshape(predictions[c("User", "Item", "Algorithm", "Prediction")],
+                      timevar="Algorithm", idvar=c("User", "Item"),
+                      direction="wide")
ERROR: 'names' attribute [18860] must be the same length as the vector [5]
  at [<-.data.frame()
  at reshapeWide()
  at reshape()
Execution halted
```

You can obtain `predictions.csv` [here, as predictions.txt](https://github.com/lenskit/lenskit/files/26490/predictions.txt).

This error is encountered, and worked around, in my attempt to use Renjin to implement integration test verification in our project LensKit: lenskit/lenskit#814

**Comment by mjkallen** [comment]
Thanks for reporting this issue. A minimal and reproducible example is the following (using the built-in dataset `Indometh`):

```{r}
reshape(Indometh, v.names = "conc", idvar = "Subject", timevar = "time", direction = "wide")
```

which results in

```
ERROR: 'names' attribute [6] must be the same length as the vector [2]
  at [<-.data.frame()
  at reshapeWide()
  at reshape()
```

**Comment by akbertram** [comment]
According to @mjkallen the example is now running in the latest release of Renjin. @psolaimani can you add MJ's test to the master branch?

**Comment by mjkallen** [comment]
@akbertram I just ran @mdekstrand's example in Renjin 0.8.1933 and it works, but the output has some differences compared to the output from R 3.2.2:

1. the column names are different (Renjin has "Prediction.1" and "Prediction.2" whereas GNU R has "Prediction.Standard" and "Prediction.Snapshotting") which could be due to our fork of the R code being outdated;
2. the numerical accuracy is (at least visually) different. For example: GNU R will have a number 5.250357 where Renjin shows 5.25;

Bottom line: an exact comparison with the output from the latest version of GNU R will fail.

**Comment by akbertram** [comment]
The first point we need to fix: names like "Prediction.1" or "Prediction.Standard" are created when combining names from nested lists, for example `names(c(Prediction = c(5,10)))` will evaluate to "Prediction.1" and "Prediction.2" while `names(c(Prediction = c(Standard=5, Snapshotting=10)))` should evaluate to "Prediction.Standard" and "Prediction.Snapshotting".

Either the names "Standard" and "Snapshotting" are incorrectly dropped in earlier step, or the names are being combined wrong, need to check.

I'm guessing the second point is related only to printing the value, but we should confirm.

**Comment by mdekstrand** [comment]
I recently upgraded Renjin, and the specific `reshape` problem I was seeing appears to have been fixed. I have re-enabled `reshape` in our tests in lenskit/lenskit#927.

**Comment by akbertram** [comment]
Great to hear!
